hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -747,41 +747,29 @@ def is_ordered(self, link_handle: str) -> bool:
 
     def get_matched_links(
         self, link_type: str, target_handles: list[str], **kwargs
-    ) -> (
-        list[str]
-        | list[list[str]]
-        | list[tuple[str, tuple[str, ...]]]
-        | tuple[int, list[str]]
-        | tuple[int, list[list[str]]]  # TODO(angelo): simplify this return type
-    ):
+    ) -> MatchedLinksResultT:
         if link_type != WILDCARD and WILDCARD not in target_handles:
             try:
                 link_handle = self.get_link_handle(link_type, target_handles)
-                if kwargs.get('cursor') is not None:
-                    return 0, [link_handle]
-                return [link_handle]
+                return None, [link_handle]
             except AtomDoesNotExist:
-                if kwargs.get('cursor') is not None:
-                    return 0, []
-                return []
+                return None, []
 
         if link_type == WILDCARD:
             link_type_hash = WILDCARD
         else:
             link_type_hash = self._get_atom_type_hash(link_type)
```

Comment:
Seems to me that this should be a responsability of _get_atom_type_hash function.
Maybe adding a simple statement on the top of that function like:
```python
if link_type == WILDCARD:
    return WILDCARD
```
Than we could simplify this methods complexity:
```python
link_type_hash = self._get_atom_type_hash(link_type)
```

Reply:
`_get_atom_type_hash` is called in different places and I'm not sure that all callers want to get `WILDCARD` as hash when `atom_type` is a `WILDCARD`.